runtime/manual/node/migrate.md

```diff
@@ -142,127 +149,217 @@ app.listen(3000, () => {
-development. In Deno, you can continue to use your existing npm scripts while
-migrating over time to [`deno task`](../tools/task_runner.md).
+Deno has its own config file, `deno.json` or `deno.jsonc`, which can be used to
+configure your project. You can use it to define tasks, dependencies, path
+mappings, and other runtime configurations.
 
-### Running npm scripts in Deno
+### Migrating npm scripts to `deno.json`
 
-One of the ways [Deno supports existing `package.json` files](./package_json.md)
-is by executing any scripts configured there with `deno task`. Consider the
-following Node.js project with a package.json and a script configured within it.
+If preferred, you can move your npm scripts over to `deno.json` to manage your
+project's runtime configuration. This allows you to run all of the necessary
+permission flags and other runtime config in one place, which can then be run
+with `deno task`.
 
-```js title="bin/my_task.mjs"
-console.log("running my task...");
+```json
+{
+  "tasks": {
+    "dev": "deno run --allow-net --allow-read --allow-env server.js"
+  }
+}
 ```
 
-```json title="package.json"
+```sh
+deno task dev
+```
+
+### Migrating npm dependencies to `deno.json`
+
+You can also migrate your dependencies over to `deno.json` to manage your
+project's dependencies. Deno supports importing dependencies from external
+package repositories, local files and urls. To import your npm dependencies, you
+can add them to the `imports` field in `deno.json`, and add the `npm:` specifier
+to the import path:
+
+```json
 {
-  "name": "test",
-  "scripts": {
-    "start": "node bin/my_task.mjs"
+  "imports": {
+    "express": "npm:express@4"
+  }
+}
+```
+
+Deno supports multiple package registries and allows you to import dependencies
+from npm, [JSR](https://jsr.io) and HTTP URLs.
+
+```json
+{
+  "imports": {
+    "express": "npm:express@4",
+    "@luca/cases": "jsr:@luca/cases@1",
+    "foo": "https://example.com/foo.ts"
   }
 }
 ```
 
-You can execute this script with Deno by running `deno task start`.
+## Optional improvements - Linting
 
-## Using and managing npm dependencies
+Deno ships with a built-in linter that is written with performance in mind. It
+can lint large projects in a few milliseconds. You can try it out on your
+project by running:
 
-Deno supports
-[managing npm dependencies through a `package.json` file](./package_json.md).
-Note that unlike using npm at the command line, you can simply run your project
-with `deno run`, and the first time your script runs, Deno will cache all the
-necessary dependencies for your application.
+```sh
+deno lint
+```
 
-Going forward, we'd recommend that you manage dependencies through
-[`deno.json`](../getting_started/configuration_file.md) instead, which supports
-other types of imports as well.
+This will lint all files in your project. When the linter detects a problem, it
+will show it in your editor and in the terminal output. An example of what that
+might look like:
```

Comment:
```suggestion
This will lint all files in your project. When the linter detects a problem, it
will show the lien in your editor and in the terminal output. An example of what that
might look like:
```